catalog/from-k8s/resource.go

```diff
@@ -151,18 +164,23 @@ func (t *ServiceResource) Upsert(key string, raw interface{}) error {
 func (t *ServiceResource) Delete(key string) error {
 	t.serviceLock.Lock()
 	defer t.serviceLock.Unlock()
+	t.doDelete(key)
+	t.Log.Info("delete", "key", key)
+	return nil
+}
```

Comment:
This function never returns an actual error.

Reply:
Yes but it's implementing the `controller.Resource` interface:
https://github.com/hashicorp/consul-k8s/blob/81026eb1a4cb2c7c3c75b03ff36eb8328a0300e1/helper/controller/resource.go#L12-L27

Reply:
got it!